R/assert_package.R

```diff
@@ -26,40 +37,70 @@ NULL
 .assert_package <- function(pkg, fn = NULL, pkg_search = "broom.helpers", boolean = FALSE) {
   # check if min version is required -------------------------------------------
   version <- .get_min_version_required(pkg, pkg_search)
+  compare <- purrr::attr_getter("compare")(version)
 
   # check installation TRUE/FALSE ----------------------------------------------
   if (isTRUE(boolean)) {
-    return(rlang::is_installed(pkg = pkg, version = version))
+    return(rlang::is_installed(pkg = pkg, version = version, compare = compare))
   }
 
   # prompt user to install package ---------------------------------------------
   rlang::check_installed(
     pkg = pkg,
     version = version,
+    compare = compare,
     reason = switch(!is.null(fn), stringr::str_glue("for `{fn}`"))
   )
   invisible()
 }
 
 #' @rdname assert_package
 #' @export
-.get_min_version_required <- function(pkg, pkg_search = "broom.helpers") {
-  if (is.null(pkg_search)) return(NULL)
-  # get min version required for a Suggested package in gtsummary
-  utils::packageDescription(
-    pkg_search,
-    fields = c("Imports", "Depends", "Suggests", "Enhances", "LinkingTo")
-  ) %>%
-    purrr::map(
-      ~stringr::str_remove_all(.x, "[\r\n]") %>%
-        stringr::str_split(pattern = stringr::fixed(",")) %>%
-        unlist() %>%
-        {.[stringr::word(.) == pkg]} %>%
-        stringr::word(start = 2, end = -1) %>%
-        stringr::str_remove_all(pattern = " ") %>%
-        stringr::str_remove_all(pattern = "^\\(>=|\\)$") %>%
-        {switch(!rlang::is_empty(.) && !is.na(.), .)}
+.get_package_dependencies <- function(
+    pkg_search = "broom.helpers",
+    remove_duplicates = FALSE,
+    lib.loc = NULL) {
+  deps <-
+    utils::installed.packages(lib.loc = lib.loc) %>%
+    tibble::as_tibble() %>%
+    dplyr::select(dplyr::all_of(
+      c("Package", "Version", "LibPath", "Imports", "Depends", "Suggests", "Enhances", "LinkingTo")
+    )) %>%
+    dplyr::rename(
+      pkg_search = "Package",
+      pkg_search_version = "Version",
+      lib_path = "LibPath"
+    )
+
+  if (!is.null(pkg_search))
+    deps <- deps %>% dplyr::filter(.data$Package %in% .env$pkg_search)
```

Comment:
Should this be `.data$pkg_search%in% .env$pkg_search`?

Reply:
You are right!!! The col was renamed just before and I have done the change too quickly.
Solved!